circle.yml

```diff
@@ -4,96 +4,138 @@
+    - run: sudo apt-get update -qq
+    # - run: sudo apt-cache show yarn
+    - run: sudo apt-get install -y -qq yarn=1.0.2-1
+    - run: sudo rm -rf /usr/local/go*
+    - run: mkdir -p ~/downloads ~/go-path
+    - run: ls "$HOME/downloads"
+    - run: ls "$ANDROID_HOME"
+
+    - run: if [[ ! -e "$HOME/downloads/go.$GOVERSION.tar.gz" ]]; then wget -nv https://storage.googleapis.com/golang/go$GOVERSION.linux-amd64.tar.gz -O $HOME/downloads/go.$GOVERSION.tar.gz ; fi
+    - run: cd $HOME && tar -xzvf downloads/go.$GOVERSION.tar.gz
+    - run: sudo mv $HOME/go /usr/local
+
+    - run: if ! $(grep -q "Revision=24.4.1" $ANDROID_HOME/tools/source.properties); then echo y | android update sdk -u -a -t "tools"; fi
+    - run: if [ ! -e $ANDROID_HOME/build-tools/23.0.2 ]; then echo y | android update sdk -u -a -t "build-tools-23.0.2"; fi
+
+    # Android Support Repository, revision 35 / Local Maven repository for Support Libraries
+    - run: if [ ! -d "/usr/local/android-sdk-linux/extras/android/m2repository/com/android/support/design/24.1.0" ]; then echo y | android update sdk --no-ui --all --filter "extra-android-m2repository"; fi
+
     # Accept android licenses
-    - mkdir -p /usr/local/android-sdk-linux/licenses
-    - if [[ ! -e "$ANDROID_HOME/ndk-bundle" ]]; then wget -nv http://dl.google.com/android/repository/android-ndk-r11c-linux-x86_64.zip -O android-ndk-r11c-linux-x86_64.zip && unzip -o android-ndk-r11c-linux-x86_64.zip -d "$ANDROID_HOME" && mv "$ANDROID_HOME/android-ndk-r11c" "$ANDROID_HOME/ndk-bundle"; fi
-    - echo "8933bad161af4178b1185d1a37fbf41ea5269c55" > /usr/local/android-sdk-linux/licenses/android-sdk-license
-    - echo "d56f5187479451eabf01fb78af6dfcb131a6481e" >> /usr/local/android-sdk-linux/licenses/android-sdk-license
+    - run: mkdir -p /usr/local/android-sdk-linux/licenses
+    - run: if [[ ! -e "$ANDROID_HOME/ndk-bundle" ]]; then wget -nv http://dl.google.com/android/repository/android-ndk-r11c-linux-x86_64.zip -O android-ndk-r11c-linux-x86_64.zip && unzip -o android-ndk-r11c-linux-x86_64.zip -d "$ANDROID_HOME" && mv "$ANDROID_HOME/android-ndk-r11c" "$ANDROID_HOME/ndk-bundle"; fi
+    - run: echo "8933bad161af4178b1185d1a37fbf41ea5269c55" > /usr/local/android-sdk-linux/licenses/android-sdk-license
+    - run: echo "d56f5187479451eabf01fb78af6dfcb131a6481e" >> /usr/local/android-sdk-linux/licenses/android-sdk-license
+
     # Install Yarn
-  cache_directories:
-    - ~/downloads
-    - ~/.cache/yarn
-    - /tmp/go-android/bin
-    - /tmp/go-android/pkg
-    - /usr/local/android-sdk-linux/tools
-    - /usr/local/android-sdk-linux/ndk-bundle
-    - /usr/local/android-sdk-linux/build-tools/23.0.2
-    - /usr/local/android-sdk-linux/extras/android/m2repository
-  override:
-    # TODO(mm): use the emulator when we figure out a consistent way of getting this working
-    # - emulator -avd circleci-android22 -no-audio -no-window:
-    #    background: true
-    #    parallel: true
-    - rm -rf ~/client/shared/node_modules
```

Comment:
removed this, since ~/client/shared/node_modules isn't cached

I wonder what the reason is for not keeping it around...